ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiWorldStateUpdater.java

```diff
@@ -687,4 +700,48 @@ public boolean isDirty() {
         && storageToClear.isEmpty()
         && codeToUpdate.isEmpty());
   }
+
+  public static class AccountConsumingMap<T> extends ConcurrentHashMap<Address, T> {
```

Comment:
I would strongly advise not to extend standard library classes unless you are very sure about what you are doing. This can create unforeseen consequences where you might break invariants of the parent class when you are not aware you are doing it. This goes double when dealing with concurrent collections. You can achieve similar behaviour using delegation instead of inheritance.

Even if you do not break any invariants now, it's an easy thing for the next developer making some simple change that break an invariant.